src/System.Private.ServiceModel/tests/Scenarios/Contract/XmlSerializer/XmlSerializerFormatTest.cs

```diff
@@ -222,4 +222,59 @@ MessageHeader header
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, client);
         }
     }
+
+    [Fact]
+    [OuterLoop]
+    public static void MessageHeader_ResponseTypeWithUsesMessageHeaderAttribute()
+    {
+        // *** SETUP *** \\
+        var binding = new BasicHttpBinding();
+        var endpointAddress = new EndpointAddress(Endpoints.HttpBaseAddress_Basic);
+        var factory = new ChannelFactory<IXmlMessageContarctTestService>(binding, endpointAddress);
+        IXmlMessageContarctTestService serviceProxy = factory.CreateChannel();
+        var input = new XmlMessageContractTestRequest("1");
+
+        try
+        {
+            // *** EXECUTE *** \\
+            XmlMessageContractTestResponse response = serviceProxy.EchoMessageResponseWithMessageHeader(input);
+
+            // *** VALIDATE *** \\
+            Assert.True(response != null);
+            Assert.Equal(input.Message, response.Message);
```

Comment:
Assert.True or Assert.False should provide a meaningful error message so we can debug failures in the lab.